### PR TITLE
Task/moko 1438 moko network

### PR DESCRIPTION
**Pull request by DenielGrin**
Added moko network

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -14,3 +14,5 @@ kotlin.native.binary.memoryModel=experimental
 kotlin.version=1.8.0
 agp.version=7.1.3
 compose.version=1.4.0-alpha01-dev954
+kotlinSerialization.version = 1.8.0
+mokoNetwork.version = 0.20.1
```
Comment (Alex009) [inline gradle.properties]: format as other declarations please

File: shared/build.gradle.kts [hunk 2]
```diff
@@ -37,13 +38,17 @@ kotlin {
                 implementation(compose.material)
                 @OptIn(org.jetbrains.compose.ExperimentalComposeLibrary::class)
                 implementation(compose.components.resources)
+                api("dev.icerock.moko:mvvm-core:0.15.0")
+                api("io.ktor:ktor-client-core:2.2.4")
+                api("dev.icerock.moko:network:0.20.1")
```
Comment (Alex009) [inline shared/build.gradle.kts]: it's should not be `api`. replace to `implementation`
also read version of moko-network from gradle properties to have single source of true

File: shared/build.gradle.kts [hunk 3]
```diff
@@ -37,13 +38,17 @@ kotlin {
...
             }
         }
         val androidMain by getting {
             dependencies {
                 api("androidx.activity:activity-compose:1.6.1")
                 api("androidx.appcompat:appcompat:1.6.1")
                 api("androidx.core:core-ktx:1.9.0")
+                api("io.ktor:ktor-client-okhttp:2.2.4")
```
Comment (Alex009) [inline shared/build.gradle.kts]: we should also add dep with darwin engine to iosMain

File: shared/build.gradle.kts [hunk 4]
```diff
@@ -53,11 +58,20 @@ kotlin {
...
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-swing:1.6.4")
             }
         }
     }
 }
 
+mokoNetwork {
+    spec("jokes") {
+        inputSpec = file("src/api/openapi.yml")
+        isInternal = false
```
Comment (Alex009) [inline shared/build.gradle.kts]: why false? should be internal

File: shared/src/commonMain/kotlin/SimpleScreen.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun SimpleScreen(
+    viewModel: SimpleViewModel = remember { SimpleViewModel() }
+) {
+    val count: String by viewModel.count.collectAsState()
+    val jokes: String by viewModel.someJoke.collectAsState()
```
Comment (Alex009) [inline shared/src/commonMain/kotlin/SimpleScreen.kt]: `jokeText`

File: shared/src/commonMain/kotlin/SimpleScreen.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            buttonText = "Click and get moko mvvm",
+            onButtonClick = viewModel::onCountClick
+        )
+
+        //MOKO NETWORK
+        TextAndButtonTemplate(
+            title = jokes.ifEmpty { "Do you need some jokes?" },
+            buttonText = "Click and get moko network",
+            onButtonClick = {
+                viewModel.apiRequest()
```
Comment (Alex009) [inline shared/src/commonMain/kotlin/SimpleScreen.kt]: without lambda creation please - use method reference. and name it `onJokePress()`

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 1 above)
Comment (DenielGrin) [reply gradle.properties]: done

File: shared/build.gradle.kts (hunk 2 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done

File: shared/build.gradle.kts (hunk 4 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done

File: shared/build.gradle.kts (hunk 3 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done

File: shared/src/commonMain/kotlin/SimpleScreen.kt (hunk 5 above)
Comment (DenielGrin) [reply shared/src/commonMain/kotlin/SimpleScreen.kt]: done

File: shared/src/commonMain/kotlin/SimpleScreen.kt (hunk 6 above)
Comment (DenielGrin) [reply shared/src/commonMain/kotlin/SimpleScreen.kt]: done

**Comment by Alex009** [comment]
i think in context of template - moko-network is optional library and we should not add it as default

**Pull request closed without merging**